### ISSUE TITLE
Lua Coroutine error when trying to create commit message

### ISSUE THREAD

**Issue by nobleach** [TimUntersberger/neogit#413]
### Description

I can stage and unstage files/hunks. When I attempt to use `cc` to create a commit message, I get a lua error.

### Neovim version

NVIM v0.9.0-dev-1640+gc0840087c-dirty
Build type: Release
LuaJIT 2.1.0-beta3

### Operating system and version

macOS 11.5

### Steps to reproduce

1. Be in a git repo
2. Stage files with Neogit
3. Use `cc` to attempt to create commit message

### Expected behavior

Expect to open new line at top of buffer for a commit message

### Actual behavior

The following Lua error is thrown:
```
Error detected while processing BufUnload Autocommands for "*":
Error executing vim.schedule lua callback: ...ck/packer/start/plenary.nvim/lua/plenary/async/async.lua:18: The coroutine failed with this message: ...pack/packer/start/neogit/lua/neogit/
lib/notification.lua:36: Vim:E937: Attempt to delete a buffer that is in use: NeogitCommitPopup
stack traceback:
        [C]: in function 'error'
        ...ck/packer/start/plenary.nvim/lua/plenary/async/async.lua:18: in function 'callback_or_next'
        ...ck/packer/start/plenary.nvim/lua/plenary/async/async.lua:45: in function <...ck/packer/start/plenary.nvim/lua/plenary/async/async.lua:44>
        [C]: in function 'wait'
        .../packer/start/nvim-tree.lua/lua/nvim-tree/git/runner.lua:137: in function '_wait'
        .../packer/start/nvim-tree.lua/lua/nvim-tree/git/runner.lua:156: in function 'run'
        ...ck/packer/start/nvim-tree.lua/lua/nvim-tree/git/init.lua:47: in function 'reload_project'
        ...ck/packer/start/nvim-tree.lua/lua/nvim-tree/git/init.lua:31: in function 'reload'
        ...m-tree.lua/lua/nvim-tree/actions/reloaders/reloaders.lua:41: in function 'reload_explorer'
        ...m/site/pack/packer/start/nvim-tree.lua/lua/nvim-tree.lua:375: in function <...m/site/pack/packer/start/nvim-tree.lua/lua/nvim-tree.lua:373>
        [C]: in function 'nvim_buf_delete'
        .../site/pack/packer/start/neogit/lua/neogit/lib/buffer.lua:102: in function 'close'
        ...m/site/pack/packer/start/neogit/lua/neogit/lib/popup.lua:63: in function 'close'
        ...m/site/pack/packer/start/neogit/lua/neogit/lib/popup.lua:220: in function 'cb'
        .../site/pack/packer/start/neogit/lua/neogit/lib/buffer.lua:328: in function <.../site/pack/packer/start/neogit/lua/neogit/lib/buffer.lua:327>
        .../packer/start/neogit/lua/neogit/lib/mappings_manager.lua:9: in function 'invoke'
        [string ":lua"]:1: in main chunk
```

### Minimal config

```Lua
vim.cmd([[set runtimepath=$VIMRUNTIME]])
vim.cmd([[set packpath=/tmp/nvim/site]])
local package_root = "/tmp/nvim/site/pack"
local install_path = package_root .. "/packer/start/packer.nvim"
local function load_plugins()
  require("packer").startup({
    {
      "wbthomason/packer.nvim",
      {
        "TimUntersberger/neogit",
        requires = {
          { "nvim-lua/plenary.nvim" },
          { "sindrets/diffview.nvim" },
        },
        config = function()
          print("loaded neogit")
          require("neogit").setup()
        end,
      },
    },
    config = {
      package_root = package_root,
      compile_path = install_path .. "/plugin/packer_compiled.lua",
      display = { non_interactive = true },
    },
  })
end
if vim.fn.isdirectory(install_path) == 0 then
  print("Installing neogit and dependencies.")
  vim.fn.system({ "git", "clone", "--depth=1", "https://github.com/wbthomason/packer.nvim", install_path })
end
load_plugins()
require("packer").sync()
```

**Comment by vinibispo** [comment]
I couldn't reproduce it, It worked successfully here, but I'm in Ubuntu.
If you can provide further details, I think I can help

**Comment by nobleach** [comment]
I'm certain the problem exists on my end. Let's close this for now since it's probably just noise.

**Comment by vinibispo** [comment]
@nobleach How did you install Neovim?
I think I have this kind of problem using null-ls, but when I reinstalled it, it worked

**Comment by nobleach** [comment]
On this machine, I've installed it via Homebrew (HEAD, not stable). And I think you're right about null-ls being the culprit. I had switched from using asdf's node install to using nvm... so many of my terminals in Tmux couldn't find a `node` executable. Thus null-ls couldn't attach to eslint/prettier/etc. Rebooting seems to have sorted it out. I could not repro on my Linux box, so I'm 99.99% sure it was just a flub. My apologies for reporting an issue.
